arrow-fx-stm/src/main/kotlin/arrow/fx/stm/STM.kt

```diff
@@ -208,22 +209,90 @@ interface STM {
    * Rerun the current transaction.
    *
    * Aborts the transaction and suspends until any of the accessed [TVar]'s changed, after which the transaction restarts.
+   *
+   * ```kotlin:ank:playground
+   * import arrow.fx.stm.atomically
+   * import arrow.fx.stm.stm
+   * import arrow.fx.coroutines.Environment
+   *
+   * suspend fun main() {
+   *   Environment().unsafeRunSync {
+   *     //sampleStart
+   *     val result = atomically {
+   *       stm { retry() } orElse { "Alternative" }
+   *     }
+   *     //sampleEnd
+   *     println("Result $result")
+   *   }
+   * }
```

Comment:
None of these examples ever needs more than the `suspend` keyword since they never actually suspend. (Except the ones for the general `STM` docs).

What happens if a suspend function run by `suspend fun main` suspends? Is there any coroutine machinery that would allow other coroutines to run and unblock that one? If not then I'd keep `Environment` so that anyone who copies this in can't shoot themselves in the foot while experimenting. But all of this is guesswork so :see_no_evil:

Reply:
Kotlin lowers `suspend fun main` into a regular `suspend` function by wrapping in `runSuspend` which on the JVM  spins until the `Contiuation` completes. So it's similar to `Enviroment().unsafeRunSync`, but I don't think we should promote the use of `unsafeRunSync` anywhere since it relies on a lock.

https://github.com/JetBrains/kotlin/blob/9adc1a6e9bd1f79ecc58988a659d093cf40f6434/compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/MainMethodGenerationLowering.kt#L75

https://github.com/JetBrains/kotlin/blob/deb416484c5128a6f4bc76c39a3d9878b38cec8c/libraries/stdlib/jvm/src/kotlin/coroutines/jvm/internal/RunSuspend.kt

Reply:
So what should we do here? Just `suspend fun` to keep it simple or use one of our runners? Also I think `STM` has no actual dependency on `arrow-fx-coroutines` atm except for docs. Maybe I should remove that entirely so we can drop the dependency entirely...

Reply:
If we can drop the dependency, then let's do that! IMO we should use `suspend fun` in the examples, to be more clear. Ideally, if cancellation ever drops in the standard library, then the need to use `Environment` should drop to almost 0%. At that point, it should be used exclusively to interop with Java APIs that do not support any other effect types. Even `CompletableFuture` is a better option than `Environment.unsafeRunSync`.

Reply:
I changed all examples except the first few because they actually use `arrow-fx-coroutines`. I also changed the dep to just core and since ank is global and includes everything anyway the few examples that need `fx-coroutines` should work just fine I think

Reply:
Awesome, just reviewed the new changes, and they look great 👏 👏 👏